apps/browser/lib/hooks/useWorkspaceUploadUpdate.ts

```diff
@@ -0,0 +1,30 @@
+import { workspaceUploadsAtom } from "@/lib/store";
+import { useWorkspaceFilesUpdate } from "@/lib/hooks/useWorkspaceFilesUpdate";
+import { useAtom } from "jotai";
+import { Workspace } from "@evo-ninja/agent-utils";
+import { InMemoryFile } from "@nerfzael/memory-fs";
+
+export const useWorkspaceUploadUpdate = () => {
+  const [workspaceUploads, setWorkspaceUploads] = useAtom(workspaceUploadsAtom);
+  const workspaceFilesUpdate = useWorkspaceFilesUpdate();
+
+  return async (workspace: Workspace, uploads?: InMemoryFile[]) => {
+    uploads = uploads || workspaceUploads;
```

Comment:
Seems a bit of a strange behavior

Reply:
This was the only way to handle the case where we setWorkspaceUploads in the same render frame calling workspaceFilesUpdate. The state variable that comes from useAtom(workspaceFilesAtom) is updated in the subsequent frame.